### PR TITLE
feat[WIP]: Wishlist

### PR DESCRIPTION
**Pull request by saadahmadhere**
This PR focuses on:
1. Add Wishlist screen.
2. Add to Wishlist feature.
3. Remove from wishlist feature.

### CODE REVIEW

**Review by Hamdrive — COMMENTED** [review]
Overall clean code written, good use of components where needed
Good work implementing reducer hook with dispatch

File: src/Components/CardHorizontal/CardHorizontal.jsx [hunk 1]
```diff
@@ -18,6 +18,16 @@ const CardHorizontal = ({ product }) => {
     dispatch({ type: "REMOVE_FROM_CART", payload: _id });
   };
 
+  const isPresentInWishlist = wishlist.find(
+    (wishlistItem) => wishlistItem._id === product._id
+  );
+
```
Comment (omjadhav85) [inline src/Components/CardHorizontal/CardHorizontal.jsx]: this function is already created in Card.js. Its better to move all these helper functions in one seperate file and then import from there

File: src/Components/Categories/CategoryItem.jsx [hunk 2]
```diff
@@ -13,7 +13,6 @@ const CategoryItem = () => {
     try {
       const categoriesData = await axios.get("/api/categories");
       dispatch({ type: "CATEGORIES", payload: categoriesData.data.categories });
```
Comment (omjadhav85) [inline src/Components/Categories/CategoryItem.jsx]: we should check api response status first. Only if 200 or 201 we should dispatch

**Review by omjadhav85 — APPROVED** [review]
Clean code and proper use of ES6 functions for checking existing items in cart and wishlist. Some suggestion regarging abstraction of helper functions. Else good to merge. Great work!

**Review by abhij1607 — COMMENTED** [review]
Very clean readable code | | Modular files | | good use of conditional EmptyWishlist Component

File: src/Components/CardHorizontal/CardHorizontal.jsx (hunk 1 above)
Comment (saadahmadhere) [reply src/Components/CardHorizontal/CardHorizontal.jsx]: resolved here:
https://github.com/saadahmadhere/elmart/pull/6/commits/7f0318c693558409d02bd8ff8ccf0343eb916664

File: src/Components/Categories/CategoryItem.jsx (hunk 2 above)
Comment (saadahmadhere) [reply src/Components/Categories/CategoryItem.jsx]: resolved here:
https://github.com/saadahmadhere/elmart/pull/6/commits/ff043b959f8db9127dca85ed3441ea33693dcab8

**Pull request merged**